### DISCUSSION TITLE
splitting & plotting the dataset

### DISCUSSION THREAD

**Discussion by anthonysirico** [pyg-team/pytorch_geometric#5555] [Q&A]
If I have a dataset that looks like this:

`dataset = CircuitDataset(43249)`

and I break it up doing:
```
ones = dataset[:10808]
zeros = dataset[10808:]
```

How can I access the values  in `ones` and `zeros`? Because if I want to plot a certain parameter like I do with dataset:
`plt.plot(dataset.data.performance`

it does not work the same way with `ones` and `zeros`, it still plots as if all the data is in ones or zeros.

How do I plot just ones performance?

**Comment by rusty1s** [comment]
Yeah, slicing of the dataset will not copy the underlying data but share it. You can do
```python
ones = dataset[:10808].copy()
zeros = dataset[10808:].copy()
```
to fix this.
